bookkeeper-dist/src/main/resources/LICENSE-all.bin.txt

```diff
@@ -0,0 +1,707 @@
+ "Software"), to  deal in  the Software without  restriction, including
+ without limitation  the rights to  use, copy, modify,  merge, publish,
+ distribute,  sublicense, and/or sell  copies of  the Software,  and to
+ permit persons to whom the Software  is furnished to do so, subject to
+ the following conditions:
+
+ The  above  copyright  notice  and  this permission  notice  shall  be
+ included in all copies or substantial portions of the Software.
+
+ THE  SOFTWARE IS  PROVIDED  "AS  IS", WITHOUT  WARRANTY  OF ANY  KIND,
+ EXPRESS OR  IMPLIED, INCLUDING  BUT NOT LIMITED  TO THE  WARRANTIES OF
+ MERCHANTABILITY,    FITNESS    FOR    A   PARTICULAR    PURPOSE    AND
+ NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ OF CONTRACT, TORT OR OTHERWISE,  ARISING FROM, OUT OF OR IN CONNECTION
+ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+------------------------------------------------------------------------------------
+For lib/protobuf-java-*.jar
+
+Copyright 2008, Google Inc.
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+
+    * Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above
+copyright notice, this list of conditions and the following disclaimer
+in the documentation and/or other materials provided with the
+distribution.
+    * Neither the name of Google Inc. nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+Code generated by the Protocol Buffer compiler is owned by the owner
+of the input file used when generating it.  This code is not
+standalone and requires a support library to be linked with it.  This
+support library is itself covered by the above license.
+
+------------------------------------------------------------------------------------
+For lib/javax.servlet-javax.servlet-api-3.1.0.jar
```

Comment:
please remove version. otherwise, it becomes really *painful* to maintain this.

Reply:
licensing can change between versions. a case in point is protobuf. We ship with protobuf 3.4.0. The license for protobuf in the this file doesn't match https://github.com/google/protobuf/blob/v3.4.0/LICENSE. There are notices for 2 c++ headers, which doesn't apply to us, but also the copyright year has changed. There are 6 years of changes we are not attributing to google.

That said, we shouldn't expect devs to remember to update notice and license each time they update a dependency. We should have a script that actually checks this stuff, which I plan on writing next week.

Reply:
I agree with matching versions and automating this. but assume if automation is not done, removing version is the most convenient way for maintenance. We can't assume that someone is going to write automate in future. we have to assume no one is doing the work, what is the best for the project to maintain.

At the minimal, I would prefer us being consistent here, either we all use specific versions, or we remove versions. That's why I prefer removing versions here.

Reply:
@sijie I've added the version to them all now. The filenames were wrong as well.

A little pain is not a bad thing. It's an incentive to actually automate this. In any case, it's much more painful to have to -1 a release candidate when everything else looks good, just because of licences. I've -1d 3 RCs (2 BK, 1 Pulsar) in as many weeks, due to bundled jars not being handled correctly.

I hope to have a script ready early next week. To be clear, the script will only verify that the LICENSE and NOTICE files reference all bundled jars. Updating the files will still be manual, as it should be.